pylint/checkers/variables.py

```diff
@@ -405,7 +405,9 @@ def _has_locals_call_after_node(stmt, scope):
     "E0601": (
         "Using variable %r before assignment",
         "used-before-assignment",
-        "Used when a local variable is accessed before its assignment.",
+        "Used when a local variable is accessed before its assignment or "
+        "accessed only after an assignment in except/finally blocks if the "
+        "try block does not return.",
```

Comment:
```suggestion
        "Emitted when a local variable is accessed before its assignment took place. "
        "Assignments in try blocks are assumed not to have occurred. Assignments "
        "in except/finally blocks are assumed to have occurred when the associated "
        "try block contains a return statement.",
```

Pretty lengthy, but this might be a bit more informative for end-users? I'm not sure..

Reply:
Assignments in try blocks *are* assumed to have occurred when evaluating statements _after any_ except/finally handling. They are merely not assumed to have occurred when evaluating the **except/finally** statement, but not until #5384 is merged, so I didn't mention here.

And the suggested version isn't quite clear about the except/finally assumptions--they are assumed to not have executed in most cases, better to explain this than the exception.

Reply:
Glad to have a wordsmith partner though! 💯   Let me know your thoughts on phrasing.

Reply:
Second try 😄 

```suggestion
        "Emitted when a local variable is accessed before its assignment took place. "
        "Assignments in try blocks are assumed not to have occurred when evaluating "
        "associated except/finally blocks. Assignments in except/finally blocks are "
        "assumed not to have occurred, except when the associated try block "
        "contains a return statement.",
```

And agreed! You're previous docstrings and comments were very clear, no doubt we'll come up with something that's spot on for this one!

Reply:
This one looks great--except that the last part just applies to except, not finally. I just want to be clear it also reflects changes from #5384, so we should decide whether to merge that one first.

Reply:
Could you rephrase the last part and then apply? I approved #5384 knowing that this will get merged before `2.13` gets out. However, it would be even better if you could copy some of the text here and add it to that PR. Not sure how much work that will be for you, but would be appreciated I think! 😄

Reply:
Yeah I've already committed my fix to your proposed change, just waiting to push until I address the comments below. Coming right up!